### PR TITLE
update controller to look up all deployments

### PR DESCRIPTION
**Pull request by petli-openshift**

### CODE REVIEW

**Review by tiwillia — APPROVED** [review]
wait

File: pkg/controller/sentryoperator/sentryoperator_controller.go [hunk 1]
```diff
@@ -101,64 +113,42 @@ func (r *ReconcileSentryOperator) Reconcile(request reconcile.Request) (reconcil
...
+			switch {
+			case depName == SENTRY_WEB_UI:
+				dep = r.deploymentForSentryWebUI(sentry)
+			case depName == SENTRY_WORKER:
+				dep = r.deploymentForSentryWorker(sentry)
+			case depName == SENTRY_CRON:
+				dep = r.deploymentForSentryCron(sentry)
+			}
+			reqLogger.Info("Creating a new Deployment.", "Deployment.Namespace", dep.Namespace, "Deployment.Name", dep.Name)
+			err = r.client.Create(context.TODO(), dep)
```
Comment (tiwillia) [inline pkg/controller/sentryoperator/sentryoperator_controller.go:134]: This seems to be creating the deployments no mater what. Are we properly checking for the existance of _each_ deployment?

File: pkg/controller/sentryoperator/sentryoperator_controller.go [hunk 2]
```diff
@@ -101,64 +113,42 @@ func (r *ReconcileSentryOperator) Reconcile(request reconcile.Request) (reconcil
...
-		reqLogger.Info("Creating a new Deployment.", "Deployment.Namespace", dep.Namespace, "Deployment.Name", dep.Name)
-		err = r.client.Create(context.TODO(), dep)
-		if err != nil {
-			reqLogger.Error(err, "Failed to create new Deployment.", "Deployment.Namespace", dep.Namespace, "Deployment.Name", dep.Name)
+	requeue := false
+	for _, depName := range allDeployments {
+		// Check if the deployment already exists, if not create a new one
+		found := &appsv1.Deployment{}
+		err = r.client.Get(context.TODO(), types.NamespacedName{Name: depName, Namespace: sentry.Namespace}, found)
+		if err != nil && errors.IsNotFound(err) {
```
Comment (petli-openshift) [inline pkg/controller/sentryoperator/sentryoperator_controller.go:121]: The dep is created only when it's not found.
It's checked here.

File: pkg/controller/sentryoperator/sentryoperator_controller.go (hunk 1 above)
Comment (tiwillia) [reply pkg/controller/sentryoperator/sentryoperator_controller.go:134]: I See, thanks

**Review by tiwillia — APPROVED** [review]
LGTM

**Pull request merged**